### PR TITLE
Upgrade to version 4.4 lts at least

### PR DESCRIPTION
**Pull request by 94noni**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Mimic https://github.com/twigphp/Twig/pull/3587 on `symfony/*`
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/HttpClient/CachingHttpClient.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ class CachingHttpClient implements HttpClientInterface, ResetInterface
     public function __construct(HttpClientInterface $client, StoreInterface $store, array $defaultOptions = [])
     {
         if (!class_exists(HttpClientKernel::class)) {
-            throw new \LogicException(sprintf('Using "%s" requires that the HttpKernel component version 4.3 or higher is installed, try running "composer require symfony/http-kernel:^4.3".', __CLASS__));
+            throw new \LogicException(sprintf('Using "%s" requires that the HttpKernel component version 4.4 or higher is installed, try running "composer require symfony/http-kernel:^4.4".', __CLASS__));
```
Comment (derrabus) [inline src/Symfony/Component/HttpClient/CachingHttpClient.php:45]: While we're at it: I find it strange that the exception message on the 5.4 branch recommends a composer command that would install version 4.4 of the component. Maybe we should drop the constraint from the command and let Composer make the best decision for us?

**Review by derrabus — APPROVED** [review]

**Review by GromNaN — APPROVED** [review]

**Comment by fabpot** [comment]
I accepted the PR on Twig as it has a limited scope, but I don't think we need to do it on Symfony. It can block some people for no good technical reasons. :-1: for me as is.

File: src/Symfony/Component/HttpClient/CachingHttpClient.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/HttpClient/CachingHttpClient.php:45]: This is indeed something we need to do

**Comment by stof** [comment]
I agree with @fabpot here. Bumping the minimum required version is only blocking some people. It won't change anything for projects already using Symfony 4.4+ (`--prefer-lowest` is not the common way to run composer)

**Comment by nicolas-grekas** [comment]
Closing as explained, comment addressed in #44102

**Comment by noniagriconomie** [comment]
@fabpot yes I understand your point, was mainly related to `As far as i know, upgrading from vX.0|1|2|3 to vX.4 is « easy » (deprecations appart)` :)

thanks for reviewing/answering!

**Pull request closed without merging**